sdk/storage/inc/blobs/blob_client.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/storage_credential.hpp"
+#include "internal/blob_client_options.hpp"
+#include "internal/protocol/blob_rest_client.hpp"
+
+#include <string>
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  class BlobClient {
+  public:
+    // connection string
+    explicit BlobClient(
+        const std::string& connectionString,
+        std::string containerName,
+        std::string blobName,
+        BlobClientOptions options = BlobClientOptions());
+
+    // shared key auth
+    explicit BlobClient(
+        const std::string& uri,
+        const StorageSharedKeyCredential& credential,
+        BlobClientOptions options = BlobClientOptions());
+
+    // token auth
+    explicit BlobClient(
+        const std::string& uri,
+        const TokenCredential& credential,
+        BlobClientOptions options = BlobClientOptions());
+
+    // anonymous/SAS/customized pipeline auth
+    explicit BlobClient(const std::string& uri, BlobClientOptions options = BlobClientOptions());
+
+    BlobClient WithSnapshot(const std::string& snapshot);
+
+    BlobProperties GetProperties(
+        const GetBlobPropertiesOptions& options = GetBlobPropertiesOptions());
+
+    FlattenedDownloadProperties Download(
+        const DownloadBlobOptions& options = DownloadBlobOptions());
+
+  private:
+    std::string m_blobUri;
+    std::string m_snapshot;
```

Comment:
There should be a field for snapshot. The snapshot is part of the blobUri.

Reply:
fixed